File: Arcade/Intro/03 - Smooth Sailing/9 - All Longest Strings/allLongestStrings.java
String[] allLongestStrings(String[] inputArray) {

    int maxSequence = 0;
    ArrayList<String> longestStrings = new ArrayList<>();
    
    for(int i = 0; i < inputArray.length; i++)
    {
        boolean foundBiggerString = maxSequence < inputArray[i].length();
        if(foundBiggerString)
        {
            //new arrayList and update the size
            longestStrings = new ArrayList<>();
            maxSequence = inputArray[i].length();
        }
        boolean stringHasSameLength = maxSequence == inputArray[i].length();
        if(stringHasSameLength)
        {
            longestStrings.add(inputArray[i]);
        }
    }
    
    //arrayList to array
    String[] result = new String[longestStrings.size()];
    result = longestStrings.toArray(result);
    
    return result;
}

File: Arcade/Intro/03 - Smooth Sailing/9 - All Longest Strings/allLongestStrings.cpp
std::vector<std::string> allLongestStrings(std::vector<std::string> inputArray) {
    
    int maxSequence = 0;
    std::vector<std::string> result;
    
    for(int i = 0; i < inputArray.size(); i++)
    {
        bool foundBiggerString = maxSequence < inputArray.at(i).length();
        if(foundBiggerString)
        {
            //reset the array and update the max sequence value
            result.resize(0);
            maxSequence = inputArray.at(i).length();
        }
        bool validString = maxSequence == inputArray.at(i).length();
        if(validString)
        {
            result.push_back(inputArray.at(i));
        }
    }
    
    return result;
}
